File: PageObjects/LoginPage.cs
using OpenQA.Selenium;

namespace PageObjects
{
    public class LoginPage
    {
        private readonly IWebDriver _driver;

        public LoginPage(IWebDriver driver)
        {
            _driver = driver;
        }

        public IWebElement Username => _driver.FindElement(By.XPath("//input[@id='user-name']"));
        public IWebElement Password => _driver.FindElement(By.XPath("//input[@id='password']"));
        public IWebElement LoginButton => _driver.FindElement(By.XPath("//input[@id='login-button']"));

        public void EnterUsername(string username)
        {
            Username.SendKeys(username);
        }

        public void EnterPassword(string password)
        {
            Password.SendKeys(password);
        }

        public void ClickLoginButton()
        {
            LoginButton.Click();
        }
    }
}

File: PageObjects/LoginPage.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class LoginPage {

    private WebDriver driver;

    public LoginPage(WebDriver driver) {
        this.driver = driver;
    }

    public void enterUsername(String username) {
        WebElement usernameField = driver.findElement(By.xpath("//input[@id='username']"));
        usernameField.sendKeys(username);
    }

    public void enterPassword(String password) {
        WebElement passwordField = driver.findElement(By.xpath("//input[@id='password']"));
        passwordField.sendKeys(password);
    }

    public void clickLoginButton() {
        WebElement loginButton = driver.findElement(By.xpath("//button[@id='login']"));
        loginButton.click();
    }

    public String getErrorMessage() {
        WebElement error = driver.findElement(By.xpath("//div[@class='error-message-container error']"));
        return error.getText();
    }
}